### ISSUE TITLE
Use Case #1 Clarification

### ISSUE THREAD

**Issue by DavidOry** [VisionEval/VE-Installer#2]
To clarify, by:

"Enable developers to run and test VisionEval in an environment comparable to what end users would have."

Do you mean that end users will be using this installation path and therefore developers can use it as well?

**Comment by jrawbits** [comment]
Use Case #1 is about developers using this setup to create a local runnable environment with the same R version and packages that would get distributed to end users.  That's very handy when someone finds a bug and you want to check it out in the same environment they're running with (and push possible fixes into that exact same environment).  Use Case #1 is essentially the same as what the current VisionEval satisfies when developers run "Install.R" (except that it doesn't commingle hidden dependencies or version differences in the developer environment).

**Comment by jrawbits** [comment]
Let me know if you have specific suggestions for revising the ReadMe for the installer builder to better explain the use case, otherwise I'll close this issue next week.

**Comment by jrawbits** [comment]
Exactly.  See "make docker" in the current installer tree (and the issue related to Docker that I just posted - there are some inefficiencies in the current Docker build).

However, we need to write down and explore use cases so we can figure out what needs to be in the images - the principal "use case family" that I see Docker excelling at would be from someone who simply wants to run a model against a set of scenarios.  The use case currently prototyped here imagines that each VE scenario is in a folder and the docker container easily runs a model on one of those folders (one container run per scenario), with the model source code in a central location in the image.  However the AASHTO/ODOT development of the "add scenario" function may have a different workflow model for how the scenarios are constructed and processed, and the Docker image should ultimately support that.

We might eventually put the GUI, Scenario Builder and Visualizer into one or more containers of their, but I'd still prefer the models to run in their own containers (which would make multi-processing a mountain of scenarios trivial through some variant of a Docker "swarm").  Such a strategy would enable us to keep the GUI image (for example) clean of the enormous model packages and their data sets.  There is a need for a much better architectural job of decoupling VisonEval-the-models from VisionEval-the-GUI -- that needs to be on our backlog in any case.

For someone who just wants to play with VisionEval interactively, Docker is absurd overkill - the current Windows installer makes it trivial to erect a functional and complete VE interactive environment in R, and the online installer provides the same ability for any other R-suppored platform (at the price of a slower source installation).  Obviously, for the source installer, we need to resolve the package hosting question see issue #4 and also [this issue I just put up on the VisionEval.org website](https://github.com/VisionEval/visioneval.github.io/issues/3#issue-391131513)).
